e2e/app/page/notebook-cell.ts

```diff
@@ -0,0 +1,152 @@
+import {ElementHandle, Frame, Page} from 'puppeteer';
+import {getPropValue} from 'utils/element-utils';
+
+export enum CellType {
+  // To append to css selector
+  Code = '.code_cell',
+  Markdown = '.text_cell',
+  Any = '',
+}
+
+export default class NotebookCell {
+
+  private readonly page: Page;
+  private iframe: Frame; // Jupyter notebook in iframe
+  private cellIndex: number;
+  private readonly cellType: CellType;
+
+
+  constructor(page: Page, cellType: CellType, cellIndex?: number) {
+    this.page = page;
+    this.cellIndex = cellIndex;
+    this.cellType = cellType;
+  }
+
+  async getIFrame(): Promise<Frame> {
+    if (this.iframe === undefined) {
+      const frame = await this.page.waitForSelector('iframe[src*="notebooks"]');
+      this.iframe = await frame.contentFrame();
+    }
+    return this.iframe;
+  }
+
+  /**
+   * Set focus in notebook cell.
```

Comment:
Does this change this container to point to a different element on the page? Would it be better to just return a different handle to another cell, rather than letting this be mutable?

For example, an issue with this is that you haven't updated `cellType` to reflect the new cell